multy_transaction/internal/bitcoin_transaction.cpp

```diff
@@ -4,5 +4,643 @@
+public:
+    BinaryDataPtr prev_transaction_hash;
+    int32_t prev_transaction_out_index = 0;
+    BinaryDataPtr prev_transaction_out_script_pubkey;
+
+    // not a property since set by Transaction or Source itself.
+    uint32_t seq = 0xffffffff;
+    BinaryDataPtr script_signature;
+    BinaryDataPtr script_witness; // serialized separately.
+
+    Amount amount; // Not serialized:
+
+    Properties properties;
+};
+
+BitcoinTransaction::BitcoinTransaction(const Account& account)
+    : TransactionBase(CURRENCY_BITCOIN, bitcoin_traits()),
+      m_account(reinterpret_cast<const BitcoinAccount&>(account)),
+      m_version(1),
+      m_is_segwit_transaction(0),
+      m_lock_time(0),
+      m_fee(new BitcoinTransactionFee)
+{
+    //    m_properties.bind_property("segwit_mode", &m_is_segwit_transaction);
+    //    m_properties.bind_property("lock_time", &m_lock_time);
+    register_properties("Fee", m_fee->properties);
+}
+
+BinaryDataPtr BitcoinTransaction::serialize() const
+{
+    BitcoinDataStream data_stream;
+
+    serialize_to_stream(&data_stream);
+
+    const BinaryData stream_content = data_stream.get_content();
+    BinaryDataPtr result;
+    throw_if_error(binary_data_clone(&stream_content, reset_sp(result)));
+    return result;
+}
+
+template <typename T>
+void BitcoinTransaction::serialize_to_stream(T* stream) const
+{
+    T& dest_stream = *stream;
+
+    dest_stream << m_version;
+    dest_stream << as_compact_size(m_sources.size());
+    for (const auto& source : m_sources)
+    {
+        dest_stream << *source;
+    }
+
+    const auto destinations = get_non_zero_destinations();
+    dest_stream << as_compact_size(destinations.size());
+    for (const auto& destination : destinations)
+    {
+        dest_stream << *destination;
+    }
+
+    dest_stream << m_lock_time;
+}
+
+bool BitcoinTransaction::is_segwit() const
+{
+    return m_is_segwit_transaction != 0
+            && std::any_of(
+                       m_sources.begin(), m_sources.end(),
+                       [](const BitcoinTransactionSourcePtr& source) -> bool {
+                           return source->script_witness
+                                   && source->script_witness->len > 0;
+                       });
+}
+
+BitcoinTransaction::Destinations
+BitcoinTransaction::get_non_zero_destinations() const
+{
+    Destinations result;
+    for (const auto& dest : m_destinations)
+    {
+        if (dest->amount > Amount(0))
+        {
+            result.push_back(dest.get());
+        }
+    }
+    return result;
+}
+
+Amount BitcoinTransaction::get_total_fee() const
+{
+    // Estimate size of the transaction based on number of sources and
+    // destinations
+    // and multiply that by "amount per byte" fee Amount.
+
+    const size_t sources_count = m_sources.size();
+    const size_t destinations_count = get_non_zero_destinations().size();
+    return (sources_count * 147 + destinations_count * 34 + 5)
+            * m_fee->get_amount_per_byte();
```

Comment:
in macOS error not function operator* unsigned long, change to (uint64_t)(sources_count * 147 + destinations_count * 34 + 5)